internal/rbd/rbd_util.go

```diff
@@ -2061,6 +2059,10 @@ func genVolFromVolIDWithMigration(
 
 // setAllMetadata set all the metadata from arg parameters on RBD image.
 func (rv *rbdVolume) setAllMetadata(parameters map[string]string) error {
+	if !rv.EnableMetadata {
```

Comment:
@pkalever same check is also required in unsetMetadata to skip RemoveMetadata call or else we might end up calling removemetadata and discard not found erro?

Reply:
I had intentionally not mentioned that in the RemoveMetadata, because if the user had enabled setmetadata at some point in time and disabled it later, the calls coming to RemoveMetadata should be handled and metadata should be removed on previous images?

Reply:
we can document the expectations WDYT?. adding check-in unsetMetadata helps in avoiding 3 calls RemoveMetadat call.

Reply:
IMHO, it is our csi responsibility to clean the metadata that it has added. Moreover, in case metadata is absent on the image, it will skip the rest of the operation to remove metadata. But that all said, I would like to leave the final decision to you!

Reply:
@nixpanic any suggestion?

Reply:
Cool, I will consider adding `if !rv.EnableMetadata{}` defence check in removeMetadata() too, Thanks guys!